packages/guardian-prover-health-check-ui/src/components/GuardianProvers.svelte

```diff
@@ -0,0 +1,287 @@
+<script lang="ts">
+  import { onDestroy, onMount } from "svelte";
+  import {
+    fetchGuardianProversFromContract,
+    type Guardian,
+  } from "../utils/fetchGuardianProversFromContract";
+  import { ethers } from "ethers";
+  import {
+    fetchGuardianProverRequests,
+    fetchLatestGuardianProverRequest,
+    fetchStats,
+    type HealthCheck,
+    type Stat,
+  } from "../utils/fetchGuardianProverStats";
+  import Checkmark from "./Checkmark.svelte";
+
+  let defaultSize: number = 10;
+  let guardianProvers: Guardian[] = [];
+  let stats: Stat[] = [];
+  let healthChecks: HealthCheck[] = [];
+  let activeId: number = 0;
+  let nextHealthCheckPage: number = 0;
+  let healthCheckPageTotal: number = 0;
+  let statsPageTotal: number = 0;
+  let nextStatsPage: number = 0;
+  let activeSubTab: string = "healthchecks";
+  let loading: boolean = false;
+  let intervals: NodeJS.Timeout[] = [];
+
+  onMount(async () => {
+    loading = true;
+    guardianProvers = await fetchGuardianProversFromContract(
+      import.meta.env.VITE_GUARDIAN_PROVER_CONTRACT_ADDRESS,
+      new ethers.providers.JsonRpcProvider(import.meta.env.VITE_RPC_URL)
+    );
+
+    await Promise.all(
+      guardianProvers.map(async (p) => {
+        const latest = await fetchLatestGuardianProverRequest(
+          import.meta.env.VITE_GUARDIAN_PROVER_API_URL,
+          p.id
+        );
+
+        p.latestHealthCheck = latest;
+        const interval = setInterval(async () => {
+          const latest = await fetchLatestGuardianProverRequest(
+            import.meta.env.VITE_GUARDIAN_PROVER_API_URL,
+            p.id
+          );
+
+          p.latestHealthCheck = latest;
+          // alive if health check is 15 seconds old
+          if (
+            latest.alive &&
+            latest.expectedAddress == latest.recoveredAddress &&
+            Date.now() / 1000 -
+              Date.parse(p.latestHealthCheck.createdAt) / 1000 >
+              15
+          ) {
+            p.alive = false;
+          } else {
+            p.alive = true;
+          }
+          guardianProvers = guardianProvers;
+        }, 12 * 1000);
+
+        intervals.push(interval);
+      })
+    );
+
+    await toggleTab(guardianProvers[0].id);
+  });
+
```

Comment:
all of this code is just boilerplate and will change, the UI has not really been fully started or designed yet. it is simply a barebones skeleton.

Reply:
> all of this code is just boilerplate and will change, the UI has not really been fully started or designed yet. it is simply a barebones skeleton.

I understand your point. However, I believe that any functions that can be optimized should still be optimized. When you come back to work on it later, it will become easier and look more professional.